core/transaction.go

```diff
@@ -233,6 +223,10 @@ var (
 	deployAccountFelt = new(felt.Felt).SetBytes([]byte("deploy_account"))
 )
 
+func errInvalidTransactionVersion(t Transaction, version *felt.Felt) error {
+	return fmt.Errorf("invalid Transaction (type: %v) verion: %v", reflect.TypeOf(t), version.Text(felt.Base10))
+}
+
```

Comment:
let's create the following error (please see error naming section in uber-go):

```go
type invalidTransactionVersionError struct {	
	t Transaction	
	v string	
}	

func (e ErrInvalidTransactionVersion) Error() string {	
	return fmt.Sprintf(""invalid Transaction (type: %v) verion: %v", reflect.TypeOf(t), version.Text(felt.Base10))
}
```

And add a test for an unsupported version which uses require/assert's `ErrorIs`/`ErrorAs`.

Reply:
We shouldn't create typed errors unless we are explicitly handling them somewhere. Checking them in tests does not warrant a typed error since they are not really "handling" the error.